File: app/src/main/java/com/harsha/data/model/ExercisePojo.java
package com.harsha.data.model;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class ExercisePojo implements Parcelable {

    @SerializedName("difficulty")
    @Expose
    private String difficulty;

    @SerializedName("duration")
    @Expose
    private String duration;

    @SerializedName("reps")
    @Expose
    private String reps;

    @SerializedName("sets")
    @Expose
    private String sets;

    @SerializedName("subtitle")
    @Expose
    private String subtitle;

    @SerializedName("description")
    @Expose
    private String description;

    @SerializedName("illustration")
    @Expose
    private Illustration illustration;

    @SerializedName("_id")
    @Expose
    private String _id;

    @SerializedName("video")
    @Expose
    private Video video;

    @SerializedName("title")
    @Expose
    private String title;

    @SerializedName("category")
    @Expose
    private String category;

    protected ExercisePojo(Parcel in) {
        difficulty = in.readString();
        duration = in.readString();
        reps = in.readString();
        sets = in.readString();
        subtitle = in.readString();
        description = in.readString();
        illustration = in.readParcelable(Illustration.class.getClassLoader());
        _id = in.readString();
        video = in.readParcelable(Video.class.getClassLoader());
        title = in.readString();
        category = in.readString();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(difficulty);
        dest.writeString(duration);
        dest.writeString(reps);
        dest.writeString(sets);
        dest.writeString(subtitle);
        dest.writeString(description);
        dest.writeParcelable(illustration, flags);
        dest.writeString(_id);
        dest.writeParcelable(video, flags);
        dest.writeString(title);
        dest.writeString(category);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<ExercisePojo> CREATOR = new Creator<ExercisePojo>() {
        @Override
        public ExercisePojo createFromParcel(Parcel in) {
            return new ExercisePojo(in);
        }

        @Override
        public ExercisePojo[] newArray(int size) {
            return new ExercisePojo[size];
        }
    };

    public String getDifficulty ()
    {
        return difficulty;
    }

    public void setDifficulty (String difficulty)
    {
        this.difficulty = difficulty;
    }

    public String getDuration ()
    {
        return duration;
    }

    public void setDuration (String duration)
    {
        this.duration = duration;
    }

    public String getReps ()
    {
        return reps;
    }

    public void setReps (String reps)
    {
        this.reps = reps;
    }

    public String getSets ()
    {
        return sets;
    }

    public void setSets (String sets)
    {
        this.sets = sets;
    }

    public String getSubtitle ()
    {
        return subtitle;
    }

    public void setSubtitle (String subtitle)
    {
        this.subtitle = subtitle;
    }

    public String getDescription ()
    {
        return description;
    }

    public void setDescription (String description)
    {
        this.description = description;
    }

    public Illustration getIllustration ()
    {
        return illustration;
    }

    public void setIllustration (Illustration illustration)
    {
        this.illustration = illustration;
    }

    public String get_id ()
    {
        return _id;
    }

    public void set_id (String _id)
    {
        this._id = _id;
    }

    public Video getVideo ()
    {
        return video;
    }

    public void setVideo (Video video)
    {
        this.video = video;
    }

    public String getTitle ()
    {
        return title;
    }

    public void setTitle (String title)
    {
        this.title = title;
    }

    public String getCategory ()
    {
        return category;
    }

    public void setCategory (String category)
    {
        this.category = category;
    }

}

File: app/src/main/java/com/harsha/data/model/ExercisePojo.kt
package com.harsha.data.model

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class ExercisePojo() :Parcelable {
    @SerializedName("difficulty")
    @Expose
    private var difficulty: String? = null

    @SerializedName("duration")
    @Expose
    private var duration: String? = null

    @SerializedName("reps")
    @Expose
    private var reps: String? = null

    @SerializedName("sets")
    @Expose
    private var sets: String? = null

    @SerializedName("subtitle")
    @Expose
    private var subtitle: String? = null

    @SerializedName("description")
    @Expose
    private var description: String? = null

    @SerializedName("illustration")
    @Expose
    private var illustration: Illustration? = null

    @SerializedName("_id")
    @Expose
    private var _id: String? = null

    @SerializedName("video")
    @Expose
    private var video: Video? = null

    @SerializedName("title")
    @Expose
    private var title: String? = null

    @SerializedName("category")
    @Expose
    private var category: String? = null

    constructor(parcel: Parcel) : this() {
        difficulty = parcel.readString()
        duration = parcel.readString()
        reps = parcel.readString()
        sets = parcel.readString()
        subtitle = parcel.readString()
        description = parcel.readString()
        illustration = parcel.readParcelable(Illustration::class.java.classLoader)
        _id = parcel.readString()
        video = parcel.readParcelable(Video::class.java.classLoader)
        title = parcel.readString()
        category = parcel.readString()
    }

    fun getDifficulty(): String? {
        return difficulty
    }

    fun setDifficulty(difficulty: String?) {
        this.difficulty = difficulty
    }

    fun getDuration(): String? {
        return duration
    }

    fun setDuration(duration: String?) {
        this.duration = duration
    }

    fun getReps(): String? {
        return reps
    }

    fun setReps(reps: String?) {
        this.reps = reps
    }

    fun getSets(): String? {
        return sets
    }

    fun setSets(sets: String?) {
        this.sets = sets
    }

    fun getSubtitle(): String? {
        return subtitle
    }

    fun setSubtitle(subtitle: String?) {
        this.subtitle = subtitle
    }

    fun getDescription(): String? {
        return description
    }

    fun setDescription(description: String?) {
        this.description = description
    }

    fun getIllustration(): Illustration? {
        return illustration
    }

    fun setIllustration(illustration: Illustration?) {
        this.illustration = illustration
    }

    fun get_id(): String? {
        return _id
    }

    fun set_id(_id: String?) {
        this._id = _id
    }

    fun getVideo(): Video? {
        return video
    }

    fun setVideo(video: Video?) {
        this.video = video
    }

    fun getTitle(): String? {
        return title
    }

    fun setTitle(title: String?) {
        this.title = title
    }

    fun getCategory(): String? {
        return category
    }

    fun setCategory(category: String?) {
        this.category = category
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(difficulty)
        parcel.writeString(duration)
        parcel.writeString(reps)
        parcel.writeString(sets)
        parcel.writeString(subtitle)
        parcel.writeString(description)
        parcel.writeParcelable(illustration, flags)
        parcel.writeString(_id)
        parcel.writeParcelable(video, flags)
        parcel.writeString(title)
        parcel.writeString(category)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<ExercisePojo> {
        override fun createFromParcel(parcel: Parcel): ExercisePojo {
            return ExercisePojo(parcel)
        }

        override fun newArray(size: Int): Array<ExercisePojo?> {
            return arrayOfNulls(size)
        }
    }
}